### PR TITLE
DOC: improved the docstring of pandas.Series.clip_upper improved

### PR DESCRIPTION
**Pull request by jalazbe**
(pandas_dev) C:\Users\Jaime\repos\MadrisSprintPandas\pandas\scripts>python validate_docstrings.py pandas.Series.clip_upper

################################################################################
################################################################################

Return copy of the input with values above given value(s) truncated.

It truncates values above a certain threshold. Threshold can be a single
value or an array, in the latter case it performs the truncation
element-wise.

Parameters
----------
threshold : float or array_like
    Maximum value allowed. All values above threshold will be
    set to this value.
axis : int or string axis name, optional
    Align object with threshold along the given axis.
inplace : boolean, default False
    Whether to perform the operation in place on the data.

See Also
--------
clip : Return copy of input with values below/above thresholds truncated.
clip_lower : Return copy of input with values below given thresholds.

Returns
-------
clipped : same type as input

Examples
--------
>>> s = pd.Series([1,2,3,4,5,6,7])
>>> s
0    1
1    2
2    3
3    4
4    5
5    6
6    7
dtype: int64

>>> s.clip_upper(4)
0    1
1    2
2    3
3    4
4    4
5    4
6    4
dtype: int64

>>> t = [4,8,7,2,5,4,6]
>>> t
[4, 8, 7, 2, 5, 4, 6]

>>> s.clip_upper(t)
0    1
1    2
2    3
3    2
4    5
5    4
6    6
dtype: int64

################################################################################
################################################################################

Docstring for "pandas.Series.clip_upper" correct. :)

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5635,24 +5635,68 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
 
     def clip_upper(self, threshold, axis=None, inplace=False):
         """
-        Return copy of input with values above given value(s) truncated.
+        Return copy of the input with values above given value(s) truncated.
+
+        It truncates values above a certain threshold. Threshold can be a
+        single value or an array, in the latter case it performs the
+        truncation
+        element-wise.
```
Comment (datapythonista) [inline pandas/core/generic.py]: unnecessary break line.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5635,24 +5635,68 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
 
         Parameters
         ----------
         threshold : float or array_like
```
Comment (datapythonista) [inline pandas/core/generic.py]: I think we use array-like (with hyphen)

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5635,24 +5635,68 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+            Maximum value allowed. All values above threshold will be
+            set to this value.
         axis : int or string axis name, optional
             Align object with threshold along the given axis.
         inplace : boolean, default False
```
Comment (datapythonista) [inline pandas/core/generic.py]: we use `bool` instead of boolean

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5635,24 +5635,68 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
-            Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+            Whether to perform the operation in place on the data.
```
Comment (datapythonista) [inline pandas/core/generic.py]: Keep the version added

File: pandas/core/generic.py [hunk 5]
```diff
@@ -5635,24 +5635,68 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+        clip : Return input copy with values below/above thresholds truncated.
+        clip_lower : Return input copy input with values below given thresholds.
 
         Returns
         -------
         clipped : same type as input
+
+        Examples
+        --------
+        >>> s = pd.Series([1,2,3,4,5,6,7])
```
Comment (datapythonista) [inline pandas/core/generic.py]: PEP-8 requires space after commas. Probably with few values it's more compact,and still shows the behavior.

File: pandas/core/generic.py [hunk 6]
```diff
@@ -5635,24 +5635,68 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+        0    1
+        1    2
+        2    3
+        3    4
+        4    4
+        5    4
+        6    4
+        dtype: int64
+
+        >>> t = [4,8,7,2,5,4,6]
```
Comment (datapythonista) [inline pandas/core/generic.py]: same with commas

File: pandas/core/generic.py [hunk 7]
```diff
@@ -5661,18 +5705,21 @@ def clip_lower(self, threshold, axis=None, inplace=False):
         """
         Return copy of the input with values below given value(s) truncated.
 
+
+
```
Comment (datapythonista) [inline pandas/core/generic.py]: I think you didn't want to commit the changes to this pull request, did you?

File: pandas/core/generic.py [hunk 8]
```diff
@@ -5661,18 +5705,21 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
             Align object with threshold along the given axis.
         inplace : boolean, default False
-            Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+            Whether to perform the operation in place on the data.
 
         See Also
         --------
-        clip
+        clip:
```
Comment (datapythonista) [inline pandas/core/generic.py]: missing description? I think the colon requires a space in the left.

**Review by datapythonista — COMMENTED** [review]
Great docstring, added some comments.

**Comment by WillAyd** [comment]
Thanks @jalazbe !

**Pull request merged**